scylla-proxy/src/proxy.rs

```diff
@@ -0,0 +1,790 @@
+                            finish_guard:  self.finish_guard.clone(),
+                            connection_close_notifier:  connection_close_tx.subscribe(),
+                            driver_addr,
+                            real_addr: self.node.real_addr,
+                            proxy_addr: self.node.proxy_addr,
+                        };
+
+                        let (tx_request, rx_request) = mpsc::unbounded_channel::<RequestFrame>();
+                        let (tx_response, rx_response) = mpsc::unbounded_channel::<ResponseFrame>();
+                        let (tx_cluster, rx_cluster) = mpsc::unbounded_channel::<RequestFrame>();
+                        let (tx_driver, rx_driver) = mpsc::unbounded_channel::<ResponseFrame>();
+
+                        tokio::task::spawn(request_receiver
+                            .receiver_from_driver(driver_read, tx_request, self.error_propagator.clone()));
+                        tokio::task::spawn(response_receiver
+                            .receiver_from_cluster(cluster_read, tx_response, self.error_propagator.clone()));
+                        tokio::task::spawn(request_sender
+                            .sender_to_cluster(cluster_write, rx_cluster, self.error_propagator.clone()));
+                        tokio::task::spawn(response_sender
+                            .sender_to_driver(driver_write, rx_driver, self.error_propagator.clone()));
+                        tokio::task::spawn(request_processor.request_processor(
+                            rx_request,
+                            tx_driver.clone(),
+                            tx_cluster.clone(),
+                            connection_no,
+                            self.node.request_rules.clone(),
+                            connection_close_tx.clone()
+                        ));
+                        tokio::task::spawn(response_processor.response_processor(
+                            rx_response,
+                            tx_driver,
+                            tx_cluster,
+                            connection_no,
+                            self.node.response_rules.clone(),
+                            connection_close_tx.clone()
+                        ));
+                        debug!("Doorkeeper of node {} spawned workers.", self.node.real_addr);
+                        connection_no += 1;
+                    }
+                    Err(err) => {
+                        error!("Error in doorkeeper for node {}: {}", self.node.real_addr, err);
+                        let _ = self.error_propagator.send(err.into());
+                        break;
+                    }
+                }}
+                _terminate = own_terminate_notifier.recv() => break
+            }
+        }
+        debug!(
+            "Doorkeeper exits: proxy {}, node {}.",
+            self.node.proxy_addr, self.node.real_addr
+        );
+    }
+
+    async fn make_streams(
```

Comment:
It would suggest, though, that the sole action here is `accepting` some connection, whereas this function establishes a connection to a remote node, too.

Reply:
OK, I won't insist on the name change.